Source/JavaScriptCore/b3/air/AirAllocateRegistersAndStackAndGenerateCode.cpp

```diff
@@ -225,44 +243,73 @@ ALWAYS_INLINE void GenerateAndAllocateRegisters::freeDeadTmpsIfNeeded()
     }
 }
 
-ALWAYS_INLINE bool GenerateAndAllocateRegisters::assignTmp(Tmp& tmp, Bank bank, bool isDef)
+ALWAYS_INLINE bool GenerateAndAllocateRegisters::assignTmp(Tmp& tmp, Bank bank, Arg::Role role)
 {
     ASSERT(!tmp.isReg());
-    if (Reg reg = m_map[tmp].reg) {
-        ASSERT(!m_namedDefdRegs.contains(reg));
-        tmp = Tmp(reg);
+
+    auto markRegisterAsUsed = [&] (Reg reg) {
+        if (Arg::isAnyDef(role))
+            m_clobberedToClear.clear(reg);
+        // At this point, it doesn't matter if we add it to the m_namedUsedRegs or m_namedDefdRegs. 
+        // We just need to mark that we can't use it again for another tmp.
         m_namedUsedRegs.set(reg);
-        ASSERT(!m_availableRegs[bank].get(reg));
-        return true;
-    }
+    };
+
+    bool mightInterfere = m_earlyClobber.numberOfSetRegisters() || m_lateClobber.numberOfSetRegisters();
+
+    auto interferesWithClobber = [&] (Reg reg) {
+        if (!mightInterfere)
+            return false;
+        if (Arg::isAnyUse(role) && m_earlyClobber.get(reg))
+            return true;
+        if (Arg::isAnyDef(role) && m_lateClobber.get(reg))
+            return true;
+        if (Arg::activeAt(role, Arg::Phase::Early) && m_earlyClobber.get(reg))
+            return true;
```

Comment:
OK, this means that early clobber interferes with early defs. (Because we already passed `Arg::isAnyUse(role) && m_earlyClobber.get(reg)` condition. So, if it becomes true, only case is that `role` is early-def.)
Maybe should we have a comment clarifying the above?